promql/quantile.go

```diff
@@ -94,27 +95,27 @@ func bucketQuantile(q float64, buckets buckets) (float64, bool) {
 		return 0
 	})
 	if !math.IsInf(buckets[len(buckets)-1].upperBound, +1) {
-		return math.NaN(), false
+		return math.NaN(), false, false
 	}
 
 	buckets = coalesceBuckets(buckets)
-	forcedMonotonic := ensureMonotonic(buckets)
+	forcedMonotonic, fixedPrecision := ensureMonotonicAndCorrectPrecisionErrors(buckets, 1e-12)
```

Comment:
Let's put the 1e-12 into a constant at the top of this file, with some rationale. You have some good empirical data in the PR description, which might go in there. There is also a fundamental argument to be made: With a relative deviation of 1e-12, we need to have 1e12 observations in the bucket so that the change of one observation is small enough to get ignored. With the usual observation rate even of very busy services, this will hardly be reached in timeframes that matters for monitoring.